outputmanager.cpp

```diff
@@ -11,16 +11,59 @@
 
 #include "outputmanager.h"
 
+#include <algorithm>
+#include <iterator>
+
 OutputManager::OutputManager(QObject *parent) : QObject(parent)
 {
 
 }
 
-void OutputManager::handleNewData(const QByteArray &data)
+void OutputManager::operator << (const QByteArray &data)
+{
+    // problem : if a buffer ending with "\r\n" happens to be cut between '\r' and '\n'
+    //           the 2 resulting QString will be interpreted as having 2 line jumps,
+    //           instead of 1, if displayed in a QTextEdit.
+    //           To avoid this, final '\r' are removed from the received buffer,
+    //           and prepended to the next one
```

Comment:
Ok, questo in effetti spiega le stranezze sui caratteri di newline che vedevamo l'altra volta insieme.